pkg/controller/scheduler.go

```diff
@@ -119,6 +115,14 @@ func (c *Controller) advanceCanary(name string, namespace string) {
 		return
 	}
 
+	// check metric servers' availability
+	if !cd.SkipAnalysis() && (cd.Status.Phase == "" || cd.Status.Phase == flaggerv1.CanaryPhaseInitializing) {
+		if err := c.checkMetricProviderAvailability(cd); err != nil {
+			c.recordEventErrorf(cd, "Error checking metric providers: %v", err)
+			return
```

Comment:
This changes the behaviour and I see it as a major breaking change. I think we should just record the error and proceed with the initialisation. There are no metrics checks during init and failing to bootstrap the app just because of a transient network error to some external system like DataDog doesn't seem ok to me.

Reply:
absolutely right, thank you for pointing out